internal/rm/nvml_manager.go

```diff
@@ -112,6 +112,11 @@ func (r *nvmlResourceManager) getPreferredAllocation(available, required []strin
 // alignedAlloc shells out to the alignedAllocationPolicy that is set in
 // order to calculate the preferred allocation.
 func (r *nvmlResourceManager) alignedAlloc(available, required []string, size int) ([]string, error) {
+	// return all available directly if allocating all available devices
+	if len(available) == size {
```

Comment:
Looking at this again, this is not correct, because it doen't take the `required` slice into account.

I would recommend:
1. Moving this check to `getPreferredAllocation` and making the call to `r.alignedAlloc` conditional on `len(available) > size`.
2. Extending `func (r *resourceManager) distributedAlloc(available, required []string, size int) ([]string, error) {` to include a check that matches this *after*  the `candidates` have been determined.